lib/winapi/fileio.c

```diff
@@ -287,6 +287,32 @@ BOOL SetFilePointerEx (HANDLE hFile, LARGE_INTEGER liDistanceToMove, PLARGE_INTE
     }
 }
 
+BOOL SetEndOfFile (HANDLE hFile)
+{
+    // Get current position in file
+    LONG highPosDWORD = 0;
+    PLONG lpHighPosDWORD = &highPosDWORD;
+    DWORD lowPosDWORD = SetFilePointer(hFile, 0, lpHighPosDWORD, FILE_CURRENT);
```

Comment:
`SetFilePointer` is not an ideal choice as its handling of 64 bit offsets is awkward. `SetFilePointerEx` is better, but I'd prefer it this code used `NtQueryInformationFile` directly.

Reply:
I'll check that possibility out.